### PR TITLE
fix Issue 21282 mixin of AliasSeq cannot alias an expression

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/dsymbolsem.d [hunk 1]
```diff
@@ -6627,6 +6627,39 @@ void aliasSemantic(AliasDeclaration ds, Scope* sc)
         if (sc2 != sc)
             sc2.pop();
 
+        if (e)  // Try to convert Expression to Dsymbol
```
Comment (thewilsonator) [inline src/dmd/dsymbolsem.d:6630]: please merge with the identical condition and comment below.

Comment (WalterBright) [reply src/dmd/dsymbolsem.d]: The comment was wrong,  I fixed it.

**Comment by thewilsonator** [comment]
This is a regression. It should target stable.

**Comment by BorisCarvajal** [comment]
@WalterBright I was fixing this issue just now and your fix seems very overkill.
The mine is just a little change to `aliasSemantic`:
```
if (e)  // Try to convert Expression to Dsymbol
{
    if (auto te = e.isTupleExp())
         s = new TupleDeclaration(te.loc, ds.ident, cast(Objects*)te.exps);
    else
...
```
Currently you can alias a tuple but only if it encapsulated as a symbol (TupleDeclaration).
The previous change to `mixin` exposed a new code path where the tuple is presented as an expression to the alias code.
My fix is simple, just create a TupleDeclaration from the TupleExpression.

**Comment by WalterBright** [comment]
@BorisCarvajal where is the mixin be expanded? Anyhow, create a PR for your version, and we'll compare.

**Comment by WalterBright** [comment]
Buildkite has ANOTHER Heisenbug:
```
[boostrap] Downloading compiler http://downloads.dlang.org/releases/2.x/2.088.0/dmd.2.088.0.linux.tar.xz
--
  | curl: (56) Recv failure: Connection reset by peer
  | xz: (stdin): Unexpected end of input
  | tar: Unexpected EOF in archive
  | tar: Unexpected EOF in archive
  | tar: Error is not recoverable: exiting now
  | ERROR: bootstrapping failed.
  | 🚨 Error: The command exited with status 1
```
https://issues.dlang.org/show_bug.cgi?id=21284

Is it possible we can get some Test Suite action on these problems? It's really not good enough to have to keep restarting the whole shebang.

https://issues.dlang.org/buglist.cgi?bug_status=NEW&bug_status=ASSIGNED&bug_status=REOPENED&keywords=TestSuite%2C%20&keywords_type=allwords&list_id=233345&query_format=advanced

**Comment by BorisCarvajal** [comment]
> where is the mixin be expanded?

I think in typesem.d->resolve()->visitMixin()

Gonna push it soon.

**Comment by thewilsonator** [comment]
That looks like a network error, good luck eliminating them entirely.

**Comment by WalterBright** [comment]
> That looks like a network error, good luck eliminating them entirely.

A number of them have been successfully fixed with the method: for network errors, sleep for a few seconds, then try it again.

The problem with not fixing these is the number of tests we have that are dependent on the network is getting so large it becomes a virtual certainty that it will fail somewhere and we can't get a clean test run.

File: test/compilable/test21282.d [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+tuple(func)
+---
+*/
+// https://issues.dlang.org/show_bug.cgi?id=21282
+
+template I(T...) { alias I = T; }
+
+template Bug(T...) {
+        alias Bug = I!(T[0]);
+        //alias Bug = mixin("I!(T[0])");
```
[the remark this answers is not in the document]
Comment (WalterBright) [reply test/compilable/test21282.d:13]: ak, you're right

**Comment by BorisCarvajal** [comment]
my pull #11814

File: test/compilable/test21282.d (hunk 2 above)
[the remark this answers is not in the document]
Comment (BorisCarvajal) [reply test/compilable/test21282.d:13]: found it when I was stealing the test case.

**Comment by WalterBright** [comment]
Closing in favor of #11814

**Pull request closed without merging**